timely_beliefs/beliefs/classes.py

```diff
@@ -544,8 +572,21 @@ def apply_belief_timing_filters(q):
                 ),
             )
 
+        # apply fast-track most recent event|belief only approach
+        if most_recent_event_only:
+            if most_recent_belief_only:
+                q = q.order_by(cls.event_start.desc(), cls.belief_horizon.desc()).limit(
```

Comment:
You want the smallest belief horizon, so its order should be ascending.

```suggestion
                q = q.order_by(cls.event_start.desc(), cls.belief_horizon.asc()).limit(
```